daemon/cmd/state.go

```diff
@@ -439,43 +444,44 @@ func (d *Daemon) allocateIPsLocked(ep *endpoint.Endpoint) (err error) {
 func (d *Daemon) initRestore(restoredEndpoints *endpointRestoreState) chan struct{} {
 	bootstrapStats.restore.Start()
 	var restoreComplete chan struct{}
+	var clusterSyncComplete chan struct{}
 	if option.Config.RestoreState {
 		// When we regenerate restored endpoints, it is guaranteed tha we have
 		// received the full list of policies present at the time the daemon
 		// is bootstrapped.
 		restoreComplete = d.regenerateRestoredEndpoints(restoredEndpoints)
-		go func() {
-			<-restoreComplete
-		}()
-
-		go func() {
-			if d.clientset.IsEnabled() {
-				// Also wait for all cluster mesh to be synchronized with the
-				// datapath before proceeding.
-				if d.clustermesh != nil {
-					err := d.clustermesh.ClustersSynced(d.ctx)
-					if err != nil {
-						log.WithError(err).Fatal("timeout while waiting for all clusters to be locally synchronized")
+		if d.clientset.IsEnabled() {
+			d.jobGroup.Add(
```

Comment:
When I tested these locally (using the clustersmesh testing scripts), I didn't see these jobs running (similarly when I rain main these tasks where run).

Seems like there's potentially some issue with the jobGroup being invoked correctly?

Reply:
Do you have any docs pointers for me to run these scripts locally ?

Reply:
Unfortunately, it doesn't seem like we document them very well, I ran:

```bash
make kind-clustermesh
make kind-clustermesh-images
kind-install-cilium-clustermesh
```

Which creates two clusters and installs cilium w/ clustermesh on both.

Reply:
ok i tried creating clusters and running cilium in my WSL setup, and my agents are in crash loop, may be its not happy that i am running a 5.15 WSL ubuntu. Do you a recommendation on what OS i can test on. (may be real Ubuntu22 ?)

`iptables command: exit status 1 stderr=\"Warning: Extension CT revision 0 not supported, missing kernel module?`

Reply:
The issue here is that the registry is not registered. You can find one example [here](https://github.com/cilium/cilium/blob/main/pkg/l2announcer/l2announcer.go#L105-L138). I'd also extract this logic outside of this function, mimicking the example above.